ttnn/cpp/ttnn/operations/creation.hpp

```diff
@@ -173,11 +171,34 @@ struct FullLikeWith {
 
 struct ZerosLike : FullLikeWith<0.0f> {};
 struct OnesLike : FullLikeWith<1.0f> {};
-struct EmptyLike : FullLikeWith<0.0f> {};
 
 inline constexpr ZerosLike zeros_like{};
 inline constexpr OnesLike ones_like{};
-inline constexpr EmptyLike empty_like{};
+
+struct Empty {
+   static ttnn::Tensor invoke(
+    const ttnn::Shape& shape,
+    const DataType& dtype,
+    const Layout& layout,
+    Device* device,
+    const std::optional<MemoryConfig>& memory_config = ttnn::DRAM_MEMORY_CONFIG) {
+        return create_device_tensor(shape, dtype, layout, device, memory_config.value());
+    }
+};
+
+struct EmptyLike {
+   static ttnn::Tensor invoke(
+    const ttnn::Tensor& tensor,
+    const std::optional<DataType>& dtype = DataType::BFLOAT16,
+    const std::optional<Layout>& layout = Layout::ROW_MAJOR,
+    const std::optional<std::reference_wrapper<Device>>& device_arg = std::nullopt,
+    const std::optional<MemoryConfig>& memory_config = ttnn::DRAM_MEMORY_CONFIG) {
```

Comment:
It doesn't seem like memory_config should be optional? Seems like it's a required argument with a default value?

Reply:
In tt_eager, yes it was, 
```
Tensor empty(
    const Shape shape,
    DataType data_type = DataType::BFLOAT16,
    Layout layout = Layout::ROW_MAJOR,
    Device* device = nullptr,
    const MemoryConfig& output_mem_config = operation::DEFAULT_OUTPUT_MEMORY_CONFIG);
```
 in TTNN however, const std::optional<MemoryConfig>& memory_config seems to be used everywhere

Reply:
@ayerofieiev-tt thoughts on this?
Having it optional means someone can pass in nullopt and have it error out since we call `.value()` on it. Don't see a reason it should be optional?

Reply:
I'm ok with this staying an optional value as it makes the code more consistent with the rest of the code.    In this case we could use TT_FATAL to guard a nullopt.

Reply:
@eyonland @tt-aho , in case of empty I'll add a TT_FATAL to guard a nullopt. and in empty_like i'll use the input tensor's mem config if mem config is not passed

Reply:
@KalaivaniMCW , after thinking about this a bit more and after a conversation with @tt-aho today it is not sitting well with me that we use TT_FATAL and guard against the nullopt.   I do realize that requiring this change will ripple through and make this PR (and a few others downstream) a bit bigger but I think ultimately this the right thing to do.